index.js

```diff
@@ -15,32 +15,39 @@ const isWritable = path => {
 };
 
 module.exports = (options = {}) => {
+	const cacheDir = process.env.CACHE_DIR;
 	const {name} = options;
-	let directory = options.cwd;
+	let directory = cacheDir || options.cwd;
 
-	if (options.files) {
-		directory = commonDir(directory, options.files);
-	} else {
-		directory = directory || process.cwd();
-	}
-
-	directory = pkgDir.sync(directory);
-
-	if (directory) {
-		if (!isWritable(path.join(directory, 'node_modules'))) {
-			return undefined;
+	if (!cacheDir) {
+		if (options.files) {
+			directory = commonDir(directory, options.files);
+		} else {
+			directory = directory || process.cwd();
 		}
+		
+		directory = pkgDir.sync(directory);
 
-		directory = path.join(directory, 'node_modules', '.cache', name);
+		if (directory) {
+			if (!isWritable(path.join(directory, 'node_modules'))) {
+				return undefined;
+			}
 
-		if (directory && options.create) {
-			makeDir.sync(directory);
-		}
+			directory = path.join(directory, 'node_modules', '.cache', name);
 
-		if (options.thunk) {
-			return (...arguments_) => path.join(directory, ...arguments_);
+			if (directory) {
+				directory = path.join(directory, 'node_modules', '.cache', name);
+			}
```

Comment:
This logic makes no sense to me.

Reply:
I rewrote the logic based on the master branch copy.